src/scheduler/data_types.h

```diff
@@ -371,6 +371,7 @@ struct queue_info
 	unsigned is_prime_queue:1;	/* only run jobs in primetime */
 	unsigned is_nonprime_queue:1;	/* only run jobs in nonprimetime */
 	unsigned has_nodes:1;		/* does this queue have nodes assoc with it */
+	unsigned has_partition:1;	/* does this queue has partition assoc with it */
```

Comment:
Is this needed?  Can't you just check qinfo->partition != NULL?

Reply:
I initially added it just to make it consistent with has_nodes. I have removed this and making use of qinfo->partition itself wherever required.